infra/src/main/scala/scaladex/infra/storage/local/LocalStorageRepo.scala

```diff
@@ -28,26 +31,60 @@ class LocalStorageRepo(dataPaths: DataPaths) extends LocalStorageApi {
 
   override def getAllProjectSettings(): Map[Project.Reference, Project.Settings] = {
     val fileContent = Files
-      .readAllLines(dataPaths.liveProjects)
+      .readAllLines(projectSettings)
       .toArray
       .mkString("")
     parser.decode[Map[Project.Reference, Project.Settings]](fileContent).toTry.get
   }
 
-  override def saveAllProjectSettings(settings: Map[Project.Reference, Project.Settings]): Unit = {
-    val liveDir = dataPaths.liveProjects.getParent
-    if (!Files.isDirectory(liveDir)) {
-      Files.createDirectory(liveDir)
-    }
-
+  override def saveAllProjectSettings(settings: Map[Project.Reference, Project.Settings]): Unit =
     Files.write(
-      dataPaths.liveProjects,
+      projectSettings,
       Printer.noSpaces.print(settings.asJson).getBytes(StandardCharsets.UTF_8)
     )
+
+  override def createTempFile(content: String, prefix: String, suffix: String): Path = {
+    val tempFile = Files.createTempFile(temp, prefix, suffix)
+    Files.write(tempFile, content.getBytes(StandardCharsets.UTF_8))
```

Comment:
hope that `Files` close it's own resource. Otherwise I have seen people writing this method like this 
```
  final def writeFile(file: AbsolutePath, content: String): Try[Unit] =
    Try {
      val bw = new BufferedWriter(new FileWriter(file.toFile))
      bw.write(content)
      bw.close()
    }
```

Reply:
Yes `Files.write` manage the resource itself.